### PR TITLE
[CINFRA-656] Maintenance drop shard on leader + Supervision wait for Current

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose
This PR does two things:
1. It adds dropping a shard using the maintenance via the Document state leader. 
2. It adds an additional check to the supervision to wait for a collection to appear in Current.

### CODE REVIEW

File: arangod/Cluster/Maintenance.cpp [hunk 1]
```diff
@@ -466,8 +466,10 @@ static void handleLocalShard(
     if (replicationVersion != replication::Version::TWO) {
```
Comment (apetenchea) [inline arangod/Cluster/Maintenance.cpp:466]: Activating this PR would mean an `|| isLeading` inside this conditional, as far as I see.

**Review by apetenchea — APPROVED** [review]
LGTM! I get that it's currently blocked, but I think it's ok to merge it if you want. Might make merge conflicts easier to fix along the way.

File: arangod/Cluster/Maintenance.cpp (hunk 1 above)
Comment (maierlars) [reply arangod/Cluster/Maintenance.cpp:466]: 😵‍💫 why is it not in this PR but was in a different one...

**Review by apetenchea — APPROVED** [review]

**Comment by maierlars** [comment]
Last error is unrelated to this PR.

**Pull request merged**